document/CoreFormat.mdk

```diff
@@ -57,6 +57,83 @@ nor are hyphen, underscore, or tilde.
 * In the host component, registered names and hexadecimal addresses use lower-case.
 ~
 
+## URI base id properties { #uriBaseIdProperties }
+
+Certain objects in this specification which have a URI-valued property (&sect;[#uriProperties])
+also have a property that is described as being a "URI base id".
+The value of such a property is a string which indirectly specifies the base URI for
+the file whose location is specified in the corresponding URI-valued property by a relative URI.
+If the URI-valued property contains an absolute URI, the URI base id property shall be absent.
+
+If the consumer of the log file requires an absolute URI (for example, to display
+the specified file to a user), then the consumer must have the necessary information
+to resolve the value of the URI base id property to an absolute URI, which can then be combined
+with the relative URI stored in the URI-valued property.
+
+The value of a URI base id property may be any string; it need not have any particular
+syntax or follow any particular naming convention.
+The tool that produces the log file and any systems that consume the log file must agree on
+the meanings of any values for the URI base id property that appear in the log file.
+
+
+~ IsoNumberedExample { @IsoNumberedExample=1 }
+In this example, the analysis tool has set the URI-valued property `result.resultFile.uri`
+(&sect;[#physicalLocation-uri]) to the relative URI of the file in which the result was detected.
+The tool has also set the value of the URI base id property `result.resultFile.uriBaseId`
+(&sect;[#physicalLocation-uriBaseId]) to `"src"`.
+By a convention agreed upon between the analysis tool and the log file consumers,
+this indicates that the relative URI is expressed relative to the root of the source tree
+in which the file appears.
+
+```
+"results": [
+  {
+    "resultFile": {
+      "uri": "drivers/video/hidef/driver.c",
+      "uriBaseId": "src"
```

Comment:
We can certainly talk about it. This was a considered choice. The thing is, these values aren't intended to be "expanded" so much as "looked up". Those substitution syntaxes make sense in the context of a larger string (%SDXROOT%\foo\bar.cs), where you have to be able to tell that the funny-looking thing isn't part of the path. Here, we've cleverly separated the "key" out from the larger string and so (as the text above says), it "need not conform to any particular syntax or follow any particular naming convention".

Thoughts?